src/sonarcloud.ts

```diff
@@ -102,17 +102,21 @@ export const makeSonarcloudAPICall = async (
   const requestInit = {
     method,
     headers: {
-      Authorization: `basic ${Buffer.from(sonarcloudApiToken, "utf8").toString(
+      Authorization: `Bearer ${Buffer.from(sonarcloudApiToken, "utf8").toString(
         "base64"
       )}`,
     },
   };
   if (method === "post") {
-    const response = await fetch(url, requestInit);
+    const response = (await fetch(url, requestInit).catch((error) => {
+      logger.error("ENGEXPUTILS015", { error });
+    })) ?? { status: 500 };
     if (response.status === 204) {
       return { success: true };
+    } else if (response.status === 500) {
+      return { success: false };
     }
-    const responseParsed = await response.json();
+    const responseParsed = await (response as Response).json();
     return checkResponse(responseParsed);
```

Comment:
Personally I find mixing the `Promise` API with `async`/`await` quite hard to think through.  If you've got a `.catch` in there that implies that the `await` would otherwise throw an error, so `try...catch` should work. Is there a reason not to do something like this?  It avoids creating that dummy `{status: 500}` which just feels off.

```suggestion
  try {
    const response = await fetch(url, requestInit);
    if (response.status === 204) {
      return { success: true };
    } else if (response.status === 500) {
      return { success: false };
    } else {
      const responseParsed = await (response as Response).json();
      return checkResponse(responseParsed);
    }
  } catch (error) {
    logger.error("ENGEXPUTILS016", { error });
    return { success: false };
  }
```

Might be worth moving the response parsing outside the `try` block just so it can't log the wrong message if that fails, but avoiding that intermediate fake response seems a good idea to me.